src/test/java/com/socialmedia/controller/ApplicationUserControllerTest.java

```diff
@@ -25,16 +30,65 @@
     @Autowired
     private MockMvc mockMvc;
 
+    @Autowired
+    private ObjectMapper mapper;
+
     @MockBean
-    private UserService userService;
+    private Principal principal;
 
     @Test
-    public void starterTestForCoverAllIntegration() throws Exception{
-        ApplicationUser userDummy = new ApplicationUser();
-        String username = "user";
-        when(userService.getUser(username)).thenReturn(Optional.of(userDummy));
+    public void getCurrentUserShouldBlockRequestWithoutAuthentication() throws Exception{
 
         mockMvc.perform(get("/api/v1/users/current"))
-                .andExpect(status().isForbidden());
+            .andExpect(status().isForbidden());
+    }
+
+    @Test
+    @WithMockUser(username = "testUser")
+    public void getCurrentUserShouldReturnUserObjectWithSuccessResponseCode() throws Exception{
+
+        ApplicationUser applicationUser = ApplicationUser.builder()
+            .username("testUser")
+            .email("test@test.com")
+            .firstName("Tyler").lastName("Durden")
+            .birthDate(659998800000L)
+            .avatar("http://avataurl.com")
+            .refreshToken("034daREFRESHTOKENf341fd")
+            .forgotPasswordToken("41fdFORGOTPASSWORDTOKEN093wgs")
+            .openAccount(true)
+            .build();
+
+        when(principal.getName()).thenReturn(applicationUser.getUsername());
```

Comment:
why have you used mock principal bean? Using annotation @WithMockUser should do the trick with auth and injecting principal into controller method

Reply:
I need this not for authentication, but for getting username in order to find user from database

Reply:
Fixed